offpunk: init at 1.4

DamienCassou opened the pull request:
###### Description of changes

> A command-line and offline-first smolnet browser/feed reader for Gemini, Gopher, Spartan and 
> Web by [Ploum](https://ploum.net/).
> The goal of Offpunk is to be able to synchronize your content once (a day, a week, a month)
> and then browse/organize it while staying disconnected.

https://notabug.org/ploum/offpunk

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

DamienCassou committed:
offpunk: init at 1.4

DamienCassou commented:
Result of `nixpkgs-review pr 179888` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

AndersonTorres commented on pkgs/applications/networking/browsers/offpunk/default.nix:
```diff
@@ -0,0 +1,46 @@
+{stdenv, testVersion, makeWrapper, python3, lib, fetchFromGitea, timg, less, offpunk, ...}:
+
+let
+  pythonDependencies = (with python3.pkgs; [
```
Useless parentheses, I believe.

AndersonTorres commented on pkgs/applications/networking/browsers/offpunk/default.nix:
```diff
@@ -0,0 +1,46 @@
+  ]);
+  otherDependencies = [ timg less ];
+in
+stdenv.mkDerivation (finalAttrs: {
```
/me being curious:

This is the new way to get rid of `rec`?

AndersonTorres commented on pkgs/applications/networking/browsers/offpunk/default.nix:
```diff
@@ -0,0 +1,46 @@
+  buildInputs = [ makeWrapper ] ++ otherDependencies ++ pythonDependencies;
+
+  installPhase = ''
+    install -D ./offpunk.py $out/bin/offpunk
```
Use the `runHook` decorations.

DamienCassou replied:
If you meant `runHook preInstall` and `runHook postInstall`, it is now done. If you meant something else, please tell me.

AndersonTorres replied:
Yes, I meant that..

AndersonTorres commented on pkgs/applications/networking/browsers/offpunk/default.nix:
```diff
@@ -0,0 +1,46 @@
+{stdenv, testVersion, makeWrapper, python3, lib, fetchFromGitea, timg, less, offpunk, ...}:
```
Afair it is not recommended to use variadic listing.

DamienCassou replied:
fixed

DamienCassou commented:
Thank you very much for your feedback @AndersonTorres. I think I took it into account.

DamienCassou commented:
Result of `nixpkgs-review pr 179888` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

AndersonTorres commented:
OfBorg is failing - trying to write `homeless-shelter`.

DamienCassou commented:
@AndersonTorres wrote:
> OfBorg is failing - trying to write homeless-shelter.

yes, this is because of a passthru testVersion I added to the package. The problem is that offpunk 1.4 tried to create configuration files when asking for `--version`. This [has been fixed recently](https://notabug.org/ploum/offpunk/issues/2) but the fix isn't released yet.

I deleted the passthru test.

DamienCassou commented:
Result of `nixpkgs-review pr 179888` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

AndersonTorres commented:
This is a bit strange to me not to use the Python packaging framework from Nixpkgs. But it is working, after all.

DamienCassou commented:
> This is a bit strange to me not to use the Python packaging framework from Nixpkgs.

I agree and I initially tried to use the Python packaging framework. But offpunk comes with one Python file only and no framework configuration. It was easier to bypass the framework than to use it and disable the automatic stuff that weren't compatible. My lack of understanding of Python packaging also played a big role in this decision :-).

fabianhjr commented:
> error: Function called without required argument "testVersion" at /home/fabian/.cache/nixpkgs-review/pr-180835/nixpkgs/pkgs/applications/networking/browsers/offpunk/default.nix:10

Investigating an issue around a missing argument.

fabianhjr commented:
Argument seems unused, could be removed.
